arcade/types/rect.py

```diff
@@ -0,0 +1,511 @@
+        return (int(self.left), int(self.right), int(self.bottom), int(self.top))
+
+    @classmethod
+    def from_kwargs(cls, **kwargs: AsFloat) -> Rect:
+        """Creates a new Rect from keyword arguments. Throws ValueError if not enough are provided.
+
+        Expected forms are:
+        * LRBT (providing `left`, `right`, `bottom`, and `top`)
+        * LBWH (providing `left`, `bottom`, `width`, and `height`)
+        * XYWH (providing `x`, `y`, `width`, and `height`)
+        """
+        # Perform iteration only once and store it as a set literal
+        specified: set[str] = {k for k, v in kwargs.items() if v is not None}
+        have_lb = 'left' in specified and 'bottom' in specified
+
+        # LRBT
+        if have_lb and 'top' in specified and 'right' in specified:
+            return LRBT(kwargs["left"], kwargs["right"], kwargs["bottom"], kwargs["top"])  # type: ignore
+
+        # LBWH
+        have_wh = 'width' in specified and 'height' in specified
+        if have_wh and have_lb:
+            return LBWH(kwargs["left"], kwargs["bottom"], kwargs["width"], kwargs["height"])  # type: ignore
+
+        # XYWH
+        if have_wh and 'x' in specified and 'y' in specified:
+            return XYWH(kwargs["x"], kwargs["y"], kwargs["width"], kwargs["height"])  # type: ignore
+
+        raise ValueError("Not enough attributes defined for a valid rectangle!")
+
+    @property
+    def kwargs(self) -> RectKwargs:
+        return {"left": self.left,
+                "right": self.right,
+                "bottom": self.bottom,
+                "top": self.top,
+                "x": self.x,
+                "y": self.y,
+                "width": self.width,
+                "height": self.height}
+
+    # Since __repr__ is handled automatically by NamedTuple, we focus on
+    # human-readable spot-check values for __str__ instead.
+    def __str__(self) -> str:
+        return (
+            f"<{self.__class__.__name__} LRBT({self.left}, {self.right}, {self.bottom}, {self.top})"
+            f" XYWH({self.x}, {self.y}, {self.width}, {self.height})>")
+
+
+# Shorthand creation helpers
+
+def LRBT(left: AsFloat, right: AsFloat, bottom: AsFloat, top: AsFloat) -> Rect:
+    """Creates a new Rect from left, right, bottom, and top parameters."""
+    width = right - left
+    height = top - bottom
+    x = left + (width / 2)
+    y = bottom + (height / 2)
+    return Rect(left, right, bottom, top, width, height, x, y)
+
+
+def LBWH(left: AsFloat, bottom: AsFloat, width: AsFloat, height: AsFloat) -> Rect:
+    """Creates a new Rect from left, bottom, width, and height parameters."""
+    right = left + width
+    top = bottom + height
+    x = left + (width / 2)
+    y = bottom + (height / 2)
+    return Rect(left, right, bottom, top, width, height, x, y)
+
+
+def XYWH(x: AsFloat, y: AsFloat, width: AsFloat, height: AsFloat, anchor: Vec2 = AnchorPoint.CENTER) -> Rect:
```

Comment:
I  am not sure about the uppercase function names.

@einarf what do you think?